### PR TITLE
Added sails-hook-autoreload to sane

### LINKED ISSUE #77: Autoreload server on file change
Hello, 

it's quite annoying to have to restart sane every time we change a server file. 
Would it be possible to use the watcher already present for the client with the server files ?

There is a sails hook here providing this feature  but it does not seem to work with sane : https://github.com/sgress454/sails-hook-autoreload

I'm really new to Javascript on server and Node.js so I do not feel confident enough to work on a PR or even evaluate the amount of work it would require but I would love not going to my console to restart sane every time my server changes

### PR DESCRIPTION
**Pull request by Systho**
- npm package has been added to generated server/package.json
- autoreload config file is generated according to docker option
- autoreload config file watch most important sails dirs but some changes still requires a fresh restart. (eg. blueprints,services)
see how https://github.com/sgress454/sails-hook-autoreload evolves for tackling those issues

Closes #77

### CODE REVIEW

File: templates/server/config/_autoreload.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+var path = require('path');
+
+module.exports.autoreload = {
+  active: true,
```
Comment (kriswill) [inline templates/server/config/_autoreload.js]: I feel like this should only be active if the current env is `development`.

Comment (Globegitter) [inline templates/server/config/_autoreload.js]: Is that something to check in the settings though, or should I maybe make a PR upstream?

**Comment by Globegitter** [comment]
Thank you for the PR @Systho will definitely get merged in - just waiting to hear back from @kriswill .

**Comment by Systho** [comment]
I've followed the suggestion of @kriswill since even if sails-hook-autoreload  handle this problem in a future version it does not in the version referenced in the stack.

I guess it doesn't prevent to propose a PR in sails-hook-autoreload

**Comment by Globegitter** [comment]
One quick thing @Systho Could you please add this to https://github.com/artificialio/sane/blob/master/CHANGELOG.md

**Pull request merged**